### DISCUSSION TITLE
Return of DriverStatus value

### DISCUSSION THREAD

**Discussion by c0nwild** [elheck/Squiddrone#314] [Coding] [answered]
Hi all,
I would like your opinion on the subject of always returning the DriverStatus (particularly in parallel to a specific return value like register content or similar). 
Pro
- additional transparency when searching for source of error
- error detection and reaction possible on a lower level
- hardware malfuncion detectable
Con
- Additional code complexity
- Wider interface
- Greater testing effort

I am currently hesitant to implement Florians solution to the Com driver. This driver is not essential to basic flight operations and a communications malfunction can be detected in time on higher levels as well. 
In my opinion it is a difference, if a hardware malfunction leads to the drone crashing into the ground because of faulty position data or just ceases to communicate. This may lead to the definition of distinct safety levels for software modules...

What do you think?

**Comment by elheck** [comment, marked as answer]
"In my opinion it is a difference, if a hardware malfunction leads to the drone crashing into the ground because of faulty position data or just ceases to communicate..." and fly onto the highway because the emergency landing/off is not received.

I think communication is very critical. 
Another advantage of using the DriverStatus is knowing what part of the public facing communication fails without needing to have the domain knowledge to interpret the signs of a failure. 

In my opinion the design compromise and testing overhead is worth the additional safety.

**Reply by c0nwild** [reply]
"...because the emergency landing/off is not received." -> Interesting use case, but currently not considered. I will add it to the model to bring it into scope.

**Reply by c0nwild** [reply]
Implemented as suggested
